docs/apis/webhooks/sharepoint-webhooks-using-azure-functions.md

```diff
@@ -0,0 +1,168 @@
+# Using Azure Functions with SharePoint webhooks #
+[Azure functions](https://docs.microsoft.com/en-us/azure/azure-functions/functions-overview) offer an easy way to host your SharePoint webhooks: you can simply add your webhook C# or Javascript code via the browser and Azure will take care of the hosting and scaling of your function! This guide shows how to setup and use Azure Functions for your webhooks.
+
+## Create a an Azure Function App
+The first step you'll need to do is creating an Azure Function App, which is a special kind of Azure Web App focused on hosting Azure Functions. Navigate to [https://portal.azure.com](https://portal.azure.com), click on the "New" and search for "Function app":
+
+![Creating an Azure Function App](../../../images/webhook-azure-function0.png)
+
+Select "Function App" and complete the information needed to create the Function App:
+
+![Complte Azure Function App details](../../../images/webhook-azure-function1.png)
+
+## Create an Azure Function
+Now that the app to host the functions is ready you can continue with creating your first Azure Function by clicking on the "New Function" link:
+
+![Adding an Azure Function](../../../images/webhook-azure-function2.png)
+
+This will offer you to start your function from a template, in the case of SharePoint webhooks we will need a HTTP triggered function and since we'll be writing C# code in our sample this means we're using the **HttpTrigger-CSharp** function template. Given that SharePoint webhook services need to be anonymously callable it's important to switch the **Authorization level** to **Anonymous**.
+
+![Choosing an Azure Function template](../../../images/webhook-azure-function3.png)
+
+>**Note:**
+>Using the **GenericWebHook** template currently does not yet work for SharePoint webhooks but the SharePoint product team is aware of this problem and will address it.
+
+The result will be a "default" Azure Function written in C#
+![The default Azure Function](../../../images/webhook-azure-function4.png)
+
+In our case we want this Azure Function to behave as a SharePoint webhook service, so we'll need to implement the following in C#:
+- Return the validationtoken if specified as URL parameter to the call. This is needed as described [here](./lists/create-subscription)
+- Process the JSON webhook notification. In below sample we've opted to store the JSON in a storage queue so that an Azure Web Job can pick it up and process it asynchronously. Depending on your needs you could also process the notification directly in your webhook service
```

Comment:
I recommend stating that if the webhook service processes the notification synchronously, it needs to do so within 5 seconds (as stated in webhooks-reference-implementation.md).